src/extensions/manifestHelper.ts

```diff
@@ -0,0 +1,68 @@
+import * as clc from "cli-color";
+
+import * as refs from "./refs";
+import { Config } from "../config";
+import { InstanceSpec } from "../deploy/extensions/planner";
+import { logger } from "../logger";
+import { promptOnce } from "../prompt";
+
+/**
+ * Write a list of instanceSpecs to extensions manifest.
+ *
+ * The manifest is composed of both the extension instance list in firebase.json, and
+ * env-var for each extension instance under ./extensions/*.env
+ *
+ * @param have a list of InstanceSpec to write to the manifest
+ * @param config existing config in firebase.json
+ * @param options nonInteractive will try to do the job without asking for user input.
+ * But only when force flag is passed this will overwrite existing .env files
+ */
+export async function writeToManifest(
```

Comment:
Tbh, I probably should have added tests for this previously, but it would be great to get unit tests for writeToManifest

Reply:
I can do that in the next PR, going to submit this today to unblock my work on Tuesday :)